file_test.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"os"
+	"testing"
+	"path/filepath"
+	"io/ioutil"
+)
+
+func TestDownloadZipFile(t *testing.T) {
+	cases := []struct {
+		repoOwner   string
+		repoName    string
+		gitTag      string
+		githubToken string
+	}{
+		{"gruntwork-io", "fetch-test-public", "v0.0.1", ""},
+		{"gruntwork-io", "fetch-test-private", "v0.0.2", os.Getenv("GITHUB_OAUTH_TOKEN")},
+	}
+
+	for _, tc := range cases {
+		zipFilePath, err := downloadGithubZipFile(tc.repoOwner, tc.repoName, tc.gitTag, tc.githubToken)
+		if err != nil {
+			t.Fatalf("Failed to download file: %s", err)
+		}
+
+		if _, err := os.Stat(zipFilePath); os.IsNotExist(err) {
+			t.Fatalf("Downloaded file doesn't exist at the expected path of %s", zipFilePath)
+		}
+	}
+}
+
+func TestDownloadZipFileWithBadRepoValues(t *testing.T) {
+	cases := []struct {
+		repoOwner   string
+		repoName    string
+		gitTag      string
+		githubToken string
+	}{
+		{"https://github.com/gruntwork-io/fetch-test-public/archive/does-not-exist.zip", "MyNameIsWhat", "x.y.z", ""},
+	}
+
+	for _, tc := range cases {
+		_, err := downloadGithubZipFile(tc.repoOwner, tc.repoName, tc.gitTag, tc.githubToken)
+		if err == nil {
```

Comment:
You may want to check for specific error codes so you know this is failing for the right reason.